modules/apps/dynamic-data-mapping/dynamic-data-mapping-form-renderer/src/main/resources/META-INF/resources/js/components/PageRenderer/EditorVariant.es.js

```diff
@@ -0,0 +1,237 @@
+/**
+ * Copyright (c) 2000-present Liferay, Inc. All rights reserved.
+ *
+ * This library is free software; you can redistribute it and/or modify it under
+ * the terms of the GNU Lesser General Public License as published by the Free
+ * Software Foundation; either version 2.1 of the License, or (at your option)
+ * any later version.
+ *
+ * This library is distributed in the hope that it will be useful, but WITHOUT
+ * ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or FITNESS
+ * FOR A PARTICULAR PURPOSE. See the GNU Lesser General Public License for more
+ * details.
+ */
+
+import ClayLayout from '@clayui/layout';
+import classNames from 'classnames';
+import domAlign from 'dom-align';
+import React, {useContext, useLayoutEffect, useRef} from 'react';
+
+import {DND_ORIGIN_TYPE, useDrop} from '../../hooks/useDrop.es';
+import {useResizeObserver} from '../../hooks/useResizeObserver.es';
+import {ParentFieldContext} from '../Field/ParentFieldContext.es';
+import Actions, {useActions} from './Actions.es';
+import * as DefaultVariant from './DefaultVariant.es';
+import {Placeholder} from './Placeholder.es';
+
+const ACTIONS_CONTAINER_OFFSET = [14, 1];
+
+export const Column = ({
+	activePage,
+	allowNestedFields,
+	children,
+	column,
+	index,
+	pageIndex,
+	rowIndex,
+}) => {
+	const parentField = useContext(ParentFieldContext);
+
+	const actionsRef = useRef(null);
+	const columnRef = useRef(null);
+
+	const {activeField, hoveredField} = useActions();
+
+	const {drop, overTarget} = useDrop({
+		columnIndex: index,
+		fieldName: column.fields[0]?.fieldName,
+		origin: DND_ORIGIN_TYPE.FIELD,
+		pageIndex,
+		parentField,
+		rowIndex,
+	});
+
+	const contentRect = useResizeObserver(columnRef);
+
+	useLayoutEffect(() => {
+		if (actionsRef.current && columnRef.current) {
+			domAlign(actionsRef.current, columnRef.current, {
+				offset: ACTIONS_CONTAINER_OFFSET,
+				points: ['bl', 'tl'],
+			});
+		}
+	}, [activeField, contentRect, hoveredField]);
```

Comment:
This is an implementation for which the Actions component should be responsible.

Reply:
Great! I moved this and other Actions related logic to Actions component :)